File: s3Tos3GlueJob.py
import sys
from awsglue.transforms import *
from awsglue.utils import getResolvedOptions
from pyspark.context import SparkContext
from awsglue.context import GlueContext
from awsglue.job import Job
print (sys.argv)
args = getResolvedOptions(sys.argv, ["JOB_NAME","input_data_path"])
sc = SparkContext()
glueContext = GlueContext(sc)
spark = glueContext.spark_session
job = Job(glueContext)
job.init(args["JOB_NAME"], args)
print (args)
# Script generated for node S3 bucket
inputDataPath =  args["input_data_path"]
S3bucket_node1 = glueContext.create_dynamic_frame.from_options(
    format_options={"quoteChar": '"', "withHeader": True, "separator": ","},
    connection_type="s3",
    format="csv",
    connection_options={
        "paths": [inputDataPath],
        "recurse": True,
    },
    transformation_ctx="S3bucket_node1",
)

# Script generated for node ApplyMapping
ApplyMapping_node2 = ApplyMapping.apply(
    frame=S3bucket_node1,
    mappings=[
        ("id", "long", "id", "long"),
        ("col1", "long", "col1", "long"),
        ("col2", "long", "col2", "long"),
        ("col3", "string", "col3", "string"),
        ("col4", "string", "col4", "string"),
    ],
    transformation_ctx="ApplyMapping_node2",
)

# Script generated for node S3 bucket
S3bucket_node3 = glueContext.write_dynamic_frame.from_options(
    frame=ApplyMapping_node2,
    connection_type="s3",
    format="csv",
    connection_options={
        "path": "s3://test-bucket-glue-ankit/output/correctness/",
        "partitionKeys": [],
    },
    transformation_ctx="S3bucket_node3",
)

job.commit()

File: s3Tos3GlueJob.scala
import com.amazonaws.services.glue.GlueContext
import com.amazonaws.services.glue.MappingSpec
import com.amazonaws.services.glue.errors.CallSite
import com.amazonaws.services.glue.util.GlueArgParser
import com.amazonaws.services.glue.util.Job
import com.amazonaws.services.glue.util.JsonOptions
import org.apache.spark.SparkContext
import scala.collection.JavaConverters._

object GlueApp {
  def main(sysArgs: Array[String]) {
    val spark: SparkContext = new SparkContext()
    val glueContext: GlueContext = new GlueContext(spark)
    // @params: [JOB_NAME]
    val args = GlueArgParser.getResolvedOptions(sysArgs, Seq("JOB_NAME").toArray)
    Job.init(args("JOB_NAME"), glueContext, args.asJava)
    // Script generated for node S3 bucket
    val S3bucket_node1 = glueContext.getSourceWithFormat(formatOptions=JsonOptions("""{"quoteChar": "\"", "withHeader": true, "separator": ","}"""), connectionType="s3", format="csv", options=JsonOptions("""{"paths": ["s3://test-bucket-glue-ankit/input/l2s2.csv"], "recurse": true}"""), transformationContext="S3bucket_node1").getDynamicFrame()

    // Script generated for node ApplyMapping
    val ApplyMapping_node2 = S3bucket_node1.applyMapping(mappings=Seq(("id", "long", "id", "long"), ("col1", "long", "col1", "long"), ("col2", "long", "col2", "long"), ("col3", "string", "col3", "string"), ("col4", "string", "col4", "string")), caseSensitive=false, transformationContext="ApplyMapping_node2")

    // Script generated for node S3 bucket
    val S3bucket_node3 = glueContext.getSinkWithFormat(connectionType="s3", options=JsonOptions("""{"path": "s3://test-bucket-glue-ankit/output/", "partitionKeys": []}"""), transformationContext="S3bucket_node3", format="json").writeDynamicFrame(ApplyMapping_node2)

    Job.commit()
  }
}
